### ISSUE TITLE
Undo cell actions

### ISSUE THREAD

**Issue by malyvsen** [fonsp/Pluto.jl#110]
Deletion & adding a cell could use ctrl+z too! I think this is a feature that everyone who isn't used to notebooks takes for granted, then they get used to Jupyter and forget it.

**Comment by fonsp** [comment]
Yes! I think that Ctrl+Z, Ctrl+Y should undo/redo your last action, whatever it was! Right now, Ctrl+Z only works within cells, and each cells maintains its own history.

**Comment by fonsp** [comment]
Related to #96

**Comment by TheCedarPrince** [comment]
I absolutely agree with this idea. I especially think this is highly important for production use. I almost lost a majority of my work today by not having the ability to "undo" a cell that I deleted. What would it take to get this implemented? Thanks @fonsp and @malyvsen !

**Comment by fonsp** [comment]
It would be c00l if both the front-end and the backend used a _transaction_-based system like Git. (Also see #152.) This isn't too different from what we already have - right now, both front-end and backend maintain an internal state, _and they communicate changes to that state_. 

So the idea is that front-end and backend _share the same_ (notebook) state, and this state can only be modified through _transactions_, each of which can be reversed deterministically.

Since this is _universal_ Ctrl+Z, keystrokes also need to be synchronized. This isn't too wild, and it would actually mean that multiple clients can do gdocs-style editing, and that unsaved changes can be backed up by the server.

As for network delays: transactions can be applied locally before being communicated with the network. This is not just useful for keystrokes - it also makes moving, adding, folding and deleting cells faster.

So like [Redux](https://redux.js.org/) but with networking.

**Comment by ederag** [comment]
@disberd Indeed, I like how [wxMaxima](https://github.com/wxMaxima-developers/wxmaxima) handles this (copied from a "tip of the day" just seen, emphasis mine) :
> Besides the **global undo** functionality that is active when the cursor is **between cells**,
wxMaxima has a **per-cell undo** function that is active if the cursor is **inside a cell**.
Pressing Ctrl+Z inside a cell can therefore been used for a fine-pitch undo
that doesn't affect latter changes made in other cells.
